main.py

```diff
@@ -1,209 +1,230 @@
+        currpos = position
         cursorUp(position)
         for i in range(0, len(arr)):
-            if(i in positionList):
+            if i in positionList:
                 cursorRight(1)
-                if tickOrCross =='X':
+                if tickOrCross == 'X':
                     setColor("Red")
-                    print("✗",end="\r\n")
-                elif tickOrCross =='T':
+                    print("✗", end="\r\n")
+                elif tickOrCross == 'T':
                     setColor("Green")
-                    print("✓",end="\r\n")       
+                    print("✓", end="\r\n")
                 else:
                     setColor(otherColor)
-                    print(tickOrCross,end="\r\n")
+                    print(tickOrCross, end="\r\n")
                 setColor("Reset")
             else:
                 cursorRight(1)
-                print(" ",end="\r\n")
-        cursorUp(len(arr)-currpos)
+                print(" ", end="\r\n")
+        cursorUp(len(arr) - currpos)
 
         sys.stdout.flush()
-        charIn=ord(sys.stdin.read(1))#Get input
-        if (charIn==13):
-            if(position==len(arr)):
-                ended=True
+        charIn = ord(sys.stdin.read(1))  # Get input
+        if charIn == 13:
+            if position == len(arr):
+                ended = True
                 break
-            if(position not in positionList):
+            if position not in positionList:
                 positionList.append(position)
             else:
                 positionList.remove(position)
-            
-
-        elif(charIn==27):#ESC pressed
-            charIn=ord(sys.stdin.read(1))
-            if(charIn==91):#[ pressed
-                charIn=ord(sys.stdin.read(1))#Get input
-                if(charIn==65):#Up
-                    if(position>0):
-                        print(" ",end="")
-                        cursorLeft(1)
-                        cursorUp(1)
-                        setColor(cursorColor)
-                        print(">",end="")
-                        setColor("Reset")
-                        cursorLeft(1)
-                        position -=1
-                elif(charIn==66):#Down
-                    if(position<len(arr)):
-                        print(" ",end="")
-                        cursorLeft(1)
-                        cursorDown(1)
-                        setColor(cursorColor)
-                        print(">",end="")
-                        setColor("Reset")
-                        cursorLeft(1)
-                        position+=1
-    cursorDown(len(arr)-position)
-    print("",end='\n\r')
-    #Set terminal style back to normal
+
+        elif charIn == 27:  # ESC pressed
+            charIn = ord(sys.stdin.read(1))
+            if charIn == 91:  # [ pressed
+                charIn = ord(sys.stdin.read(1))  # Get input
+                if charIn == 65:  # Up
+                    position = onCharUp(cursorColor, position)
```

Comment:
broke onCharUp and onCharDown into private functions, helps readability to not see blocks of 10 commands